src/pages/UserAccount/User.js

```diff
@@ -0,0 +1,152 @@
+import React, { useEffect, useState } from 'react';
+import { Link, useNavigate } from 'react-router-dom';
+import './SignUp.scss';
+
+function User({ userData, isSelectLogin }) {
+  const { title, text, url, button } = userData;
+  const [inputValue, setInputValue] = useState({ id: '', pw: '' });
+  const navigate = useNavigate();
+  const [validId, setValidId] = useState(false);
+  const [validPw, setValidPw] = useState(false);
+  const [termsAgree, setTermsAgree] = useState(false);
+
+  const handleInput = e => {
+    const { name, value } = e.target;
+    setInputValue({ ...inputValue, [name]: value });
+  };
```

Comment:
```suggestion
  const handleInput = e => {
    const { name, value } = e.target;
    setInputValue(prev => ({ ...prev, [name]: value }));
  };
```
함수형 업데이트로 구현하는 것이 더 효율적인 방법입니다!